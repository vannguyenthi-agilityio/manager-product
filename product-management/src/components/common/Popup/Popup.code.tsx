export const asPopupError = `
import Popup from '@components/Popup';

// Constants
import { POPUP_STATUS } from '@constants';

/**
 * The used to render Error popup component.
 *
 */

<Popup
  status= POPUP_STATUS.ERROR,
  content = 'Lorem ipsum dolor sit amet'
  onClick={onClosePopup} />
`;

export const asPopupWarning = `
import Popup from '@components/Popup';

// Constants
import { POPUP_STATUS } from '@constants';

/**
 * The used to render Warning popup component.
 *
 */

<Popup
  status= POPUP_STATUS.WARNING,
  content = 'Lorem ipsum dolor sit amet'
  onClick={onClosePopup} />
`;

export const asPopupSuccess = `
import Popup from '@components/Popup';

// Constants
import { POPUP_STATUS } from '@constants';

/**
 * The used to render Success popup component.
 *
 */

<Popup
  status= POPUP_STATUS.SUCCESS,
  content = 'Lorem ipsum dolor sit amet'
  onClick={onClosePopup} />
`;
