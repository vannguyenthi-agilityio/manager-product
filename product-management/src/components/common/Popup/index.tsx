import { Button, Box, Text } from '@chakra-ui/react';

// Utils
import { convertPxToRem } from '@utils/index';

// Icons
import { CloseIcon } from '@components/common/Icons';

// Constants
import { COLORS, POPUP_STATUS } from '@constants';

interface IPopupProps {
  status: POPUP_STATUS;
  content: string;
  onClosePopup: () => void;
}

const Popup = ({ status, content, onClosePopup }: IPopupProps) => {
  const textColorMapping: {
    [key: string | POPUP_STATUS]: string;
  } = {
    [`${POPUP_STATUS.ERROR}`]: `${COLORS.PINK}.600`,
    [`${POPUP_STATUS.SUCCESS}`]: `${COLORS.GREEN}.500`,
    [`${POPUP_STATUS.WARNING}`]: `${COLORS.YELLOW}.500`
  };
  return (
    <Box
      data-testid={status}
      w={convertPxToRem(434)}
      h={convertPxToRem(145)}
      p='24px 15px 24px 32px'
      bgColor={`${COLORS.GREY}.50`}
      boxShadow='lg'
      borderRadius='md'
      position='relative'
    >
      <Button
        position='absolute'
        variant='unstyled'
        onClick={onClosePopup}
        top={convertPxToRem(14)}
        right={convertPxToRem(14)}
        h='unset'
        minW='unset'
      >
        <CloseIcon />
      </Button>
      <Text
        color={textColorMapping[status]}
        textTransform='capitalize'
        fontSize='xl'
        fontWeight='light'
      >
        {status}
      </Text>
      <Text
        lineHeight='md'
        noOfLines={3}
      >
        {content}
      </Text>
    </Box>
  );
};

export default Popup;
