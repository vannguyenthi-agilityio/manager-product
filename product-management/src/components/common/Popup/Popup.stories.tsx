// Libraries
import Storybook from '@storybook/react';

// Components
import Popup from '.';

// Constants
import { POPUP_STATUS } from '@constants';

// Code
import { asPopupError, asPopupWarning, asPopupSuccess } from './Popup.code';

export default {
  title: 'Components/Popup',
  component: Popup,
  argTypes: {
    onClosePopup: { action: 'click' }
  },
  parameters: {
    componentSource: {
      language: 'typescript'
    }
  }
} as Storybook.ComponentMeta<typeof Popup>;

const Template: Storybook.ComponentStory<typeof Popup> = (args) => {
  return <Popup {...args} />;
};

export const PopupError = Template.bind({});
PopupError.args = {
  status: POPUP_STATUS.ERROR,
  content:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
};
PopupError.parameters = {
  componentSource: {
    code: asPopupError
  }
};

export const PopupWarning = Template.bind({});
PopupWarning.args = {
  status: POPUP_STATUS.WARNING,
  content: 'Lorem ipsum dolor sit amet'
};
PopupError.parameters = {
  componentSource: {
    code: asPopupWarning
  }
};

export const PopupSuccess = Template.bind({});
PopupSuccess.args = {
  status: POPUP_STATUS.SUCCESS,
  content: 'Lorem ipsum dolor sit amet'
};
PopupError.parameters = {
  componentSource: {
    code: asPopupSuccess
  }
};
