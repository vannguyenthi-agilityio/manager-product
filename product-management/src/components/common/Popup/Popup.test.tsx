// Libraries
import { render, screen } from '@testing-library/react';

// Components
import Popup from '.';

// Colors
import { POPUP_STATUS } from '@constants';

const mockProps = {
  content: 'Lorem ipsum dolor sit amet',
  onClosePopup: jest.fn()
};

describe('Popup. component', () => {
  test('should render correctly', () => {
    render(
      <Popup
        status={POPUP_STATUS.ERROR}
        {...mockProps}
      />
    );

    const popupError = screen.getByTestId('error');
    expect(popupError).toMatchSnapshot();
  });
});

describe('Popup. component', () => {
  test('should render correctly', () => {
    render(
      <Popup
        status={POPUP_STATUS.WARNING}
        {...mockProps}
      />
    );

    const popupError = screen.getByTestId('warning');
    expect(popupError).toMatchSnapshot();
  });
});

describe('Popup. component', () => {
  test('should render correctly', () => {
    render(
      <Popup
        status={POPUP_STATUS.SUCCESS}
        {...mockProps}
      />
    );

    const popupError = screen.getByTestId('success');
    expect(popupError).toMatchSnapshot();
  });
});
