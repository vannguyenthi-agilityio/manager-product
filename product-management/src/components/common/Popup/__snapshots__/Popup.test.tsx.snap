// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup. component should render correctly 1`] = `
<div
  class="css-1dt30w5"
  data-testid="error"
>
  <button
    class="chakra-button css-k21o6t"
    type="button"
  >
    <svg
      class="chakra-icon css-g7h3az"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18 6L6 18M6 6L18 18"
        stroke="#667085"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
  <p
    class="chakra-text css-mpfzx7"
  >
    error
  </p>
  <p
    class="chakra-text css-y9i0tt"
  >
    Lorem ipsum dolor sit amet
  </p>
</div>
`;

exports[`Popup. component should render correctly 2`] = `
<div
  class="css-1dt30w5"
  data-testid="warning"
>
  <button
    class="chakra-button css-k21o6t"
    type="button"
  >
    <svg
      class="chakra-icon css-g7h3az"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18 6L6 18M6 6L18 18"
        stroke="#667085"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
  <p
    class="chakra-text css-11rcpoa"
  >
    warning
  </p>
  <p
    class="chakra-text css-y9i0tt"
  >
    Lorem ipsum dolor sit amet
  </p>
</div>
`;

exports[`Popup. component should render correctly 3`] = `
<div
  class="css-1dt30w5"
  data-testid="success"
>
  <button
    class="chakra-button css-k21o6t"
    type="button"
  >
    <svg
      class="chakra-icon css-g7h3az"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18 6L6 18M6 6L18 18"
        stroke="#667085"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
  <p
    class="chakra-text css-1q0v5yv"
  >
    success
  </p>
  <p
    class="chakra-text css-y9i0tt"
  >
    Lorem ipsum dolor sit amet
  </p>
</div>
`;
